src/register/mod.rs

```diff
@@ -62,6 +62,8 @@ pub mod stimecmp;
 pub mod stimecmph;
 
 // Supervisor-Level Interrupts
+pub mod sireg;
+pub mod siselect;
```

Comment:
Since the caller is not expected to deal with siselect directly, does it make sense to not make siselect module public and only include it within sireg?

Reply:
I would argue one shouldn't be accessing things directly through sireg language in that case either since it's one piece of an index/data pair access. It seems more appropriate to wrap in an aia library dealing w/ those particular nuances.

Reply:
Yeah, I posted a comment along those lines [here](https://github.com/rivosinc/riscv/pull/10#discussion_r1014529581). I think we should make both `sireg` and `siselect` as internal implementation details. The caller should be exposed the windowed registers(`eie`, `eip`, etc.) and their implementation should deal with using the internal `sireg` and `siselect`.